thirdeye/thirdeye-pinot/src/test/java/com/linkedin/thirdeye/autoload/pinot/metrics/AutoloadPinotMetricsServiceTest.java

```diff
@@ -36,6 +37,17 @@ void afterClass() {
     super.cleanup();
   }
 
+  /**
+   * This test class needs to use a class that accesses the global singleton DAO registry. Therefore,
+   * we need to copy the local DAO registry, which is created for this test, to the global one.
+   * Otherwise, the global singleton one may reference to an arbitrary DB that is created for this
+   * test.
+   */
+  @BeforeMethod
+  void beforeMethod(){
```

Comment:
We only need this method for tests that use classes which use DAORegistry.getInstance(). Otherwise, they don't need to copy their local DAO Registry to the global singleton one.

We will work on another patch to remove the global singleton DAO Registry from the application, which is a really bad practice.